agent/bootstrap.sh

```diff
@@ -66,6 +66,15 @@ CFLAGS='-g -O0 -ftrapv' meson build \
       -Dnobody-user=nfsnobody \
       -Dnobody-group=nfsnobody
 ninja-build -C build
+
+# Let's check if the new systemd at least boots before installing it
+# As the CentOS' systemd-nspawn version is too old, we have to use QEMU
+INITRD_PATH="/boot/initramfs-$(uname -r).img"
```

Comment:
We should probably regenerate the initrd file before running the test. On the other hand, making sure that the ancient version of systemd used on CentOS still can transition to a newly built one is kind of useful too :-)

Reply:
Right, I guess I should move the dracut part before the test, as it could affect the test results.

Reply:
For it to work properly `systemd` should be installed before `initrd` is regenerated but it's dangerous because the first `daemon-reload` can crash everything as happened with Ubuntu CI in https://github.com/systemd/systemd/pull/11178 where all the useful data was lost. I'm not sure what to do about it. I guess the question here is how unsafe it is not to upgrade initrd.

Reply:
In theory, if it's possible to run the tests with `SKIP_INITRD=yes`, the whole `initrd` thing can be avoided. Could you try it to see how it goes? If it doesn't work I think it would be better to not regenerate it.

Reply:
Unfortunately, the VM dies pretty painfully... I guess the multipath root fs is one of the culprits. Let's just not regenerate the initrd and see if it's going to work.

```
[    1.312595] List of all partitions:
[    1.314035] No filesystem could mount root, tried: 
[    1.316141] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    1.319690] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 3.10.0-957.1.3.el7.x86_64 #1
[    1.322581] Hardware name: Red Hat KVM, BIOS 0.5.1 01/01/2011
[    1.324947] Call Trace:
[    1.326044]  [<ffffffff9d561e41>] dump_stack+0x19/0x1b
[    1.328059]  [<ffffffff9d55b550>] panic+0xe8/0x21f
[    1.329993]  [<ffffffff9db86761>] mount_block_root+0x291/0x2a0
[    1.332315]  [<ffffffff9db867c3>] mount_root+0x53/0x56
[    1.334499]  [<ffffffff9db86902>] prepare_namespace+0x13c/0x174
[    1.336818]  [<ffffffff9db863df>] kernel_init_freeable+0x1f8/0x21f
[    1.339424]  [<ffffffff9db85b1f>] ? initcall_blacklist+0xb0/0xb0
[    1.341981]  [<ffffffff9d54ff40>] ? rest_init+0x80/0x80
[    1.343821]  [<ffffffff9d54ff4e>] kernel_init+0xe/0x100
[    1.346068]  [<ffffffff9d574c24>] ret_from_fork_nospec_begin+0xe/0x21
[    1.348542]  [<ffffffff9d54ff40>] ? rest_init+0x80/0x80
[    1.351274] Kernel Offset: 0x1be00000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)```

Reply:
It looks like `ext4` is shipped as a separate module. In theory, it should be possible to find out what filesystems are supported "natively" and pass something like `FSTYPE=btrfs` to `make` when it sets up the environment. Having said that, the current approach looks good to me too so we should probably stop here while it still works :-)

Reply:
I'd leave it with the old initrd for now, so we have at least some debug output when something breaks. However, the `FSTYPE` shenanigans sound intriguing, will try to play with it in following days.

Reply:
Got a time to play around with using just a bootloader instead of an initrd for the QEMU, but without success, at least so far. No matter which filesystem I use (btrfs, ext*, xfs) the result is still the same. 

From what I've read, it should be enough to specify the root fs path on the kernel command line (`root=/dev/xxx`), but after several attempts using either full paths or UUIDs, I gave up. I'm not sure if I'm missing something fundamental or the initrd does something which is necessary to successfully boot the kernel.